cmd/hope/vm/image.go

```diff
@@ -0,0 +1,180 @@
+	RunE: func(cmd *cobra.Command, args []string) error {
+		vmName := args[0]
+
+		vms, err := utils.GetVMs()
+		if err != nil {
+			return err
+		}
+
+		vm, err := utils.VMSpec(vmName)
+		if err != nil {
+			return err
+		}
+
+		vmDir := path.Join(vms.Root, vm.Name)
+		log.Trace(fmt.Sprintf("Looking for VM definition in %s", vmDir))
+
+		stat, err := os.Stat(vmDir)
+		if err != nil && os.IsNotExist(err) {
+			return fmt.Errorf("VM spec directory (%s) not found", vmDir)
+		} else if err != nil {
+			return err
+		}
+
+		if !stat.IsDir() {
+			return fmt.Errorf("VM spec directory (%s) is just a file", vmDir)
+		}
+
+		packerJsonPath := path.Join(vmDir, "packer.json")
+		if _, err := os.Stat(packerJsonPath); err != nil && os.IsNotExist(err) {
+			return fmt.Errorf("VM packer file not found at path: %s", packerJsonPath)
+		} else if err != nil {
+			return err
+		}
+
+		log.Debug(fmt.Sprintf("Copying contents of %s for parameter replacement.", vmDir))
+		tempDir, err := utils.ReplaceParametersInDirectoryCopy(vmDir, vm.Parameters)
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(tempDir)
+
+		// Check caches to see if I even want to build this again.
+		tempPackerJsonPath := path.Join(tempDir, "packer.json")
+		bytes, err := ioutil.ReadFile(tempPackerJsonPath)
+		if err != nil {
+			return err
+		}
+
+		var packerSpec PackerSpec
+		if err := json.Unmarshal(bytes, &packerSpec); err != nil {
+			return err
+		}
+
+		// Packer runs out of temp dir, so output directory has to be
+		//   absolute.
+		packerOutDir := packerSpec.Builders[0].OutputDirectory
+		if !path.IsAbs(packerOutDir) {
+			return fmt.Errorf("Directory %s must be absolute;", packerOutDir)
+		}
+
+		if stat, err := os.Stat(packerOutDir); err == nil {
+			if stat.IsDir() {
+				files, err := ioutil.ReadDir(packerOutDir)
+				if err != nil {
+					return err
+				}
+
+				if len(files) != 0 {
+					filenames := []string{}
+					for _, f := range files {
+						filenames = append(filenames, f.Name())
+					}
+					log.Info(fmt.Sprintf("Files found: %s", filenames))
+					return fmt.Errorf("Directory at path %s already exists and is not empty", packerOutDir)
+				}
+			} else {
+				log.Debug(fmt.Sprintf("Will create a new directory at %s...", packerOutDir))
+			}
+		}
+
+		// Try to create a file in the same directory as the output will be.
+		// Prevents going through the whole process when the output directory
+		//   isn't writable.
+		// Seems like a no brainer for packer to do that check.
+		if err := os.MkdirAll(packerOutDir, 0755); err != nil {
+			return fmt.Errorf("Directory at path %s is not writable", packerOutDir)
+		}
+
+		allArgs := []string{"build"}
+		for _, v := range *imageCmdParameterSlice {
+			allArgs = append(allArgs, "-var", v)
+		}
+		allArgs = append(allArgs, tempPackerJsonPath)
+
+		if os.Getenv("PACKER_CACHE_DIR") == "" {
+			log.Warn("PACKER_CACHE_DIR environment variable is not set.")
+		}
+
+		if os.Getenv("PACKER_LOG") == "" {
```

Comment:
Include this and the next automatically in the packer process env. Don't require the user to provide it. Maybe set defaults and let the environment override those if provided.